pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -268,44 +268,42 @@ private void addInstanceGroupTagIfNeeded() {
    */
 
   /**
-   * Get all instance Ids.
-   *
+   * Get all instance Ids. If it's related to segment operation like assigning segments to instances, cache accessor would be used.
+   * Otherwise, direct data accessor would be used.
+   * @param segmentRelatedOperation whether it's related to segment operation
    * @return List of instance Ids
    */
   @Nonnull
-  public List<String> getAllInstances() {
-    return _cacheInstanceConfigsDataAccessor.getChildNames("/", AccessOption.PERSISTENT);
+  public List<String> getAllInstances(boolean segmentRelatedOperation) {
```

Comment:
You don't need to change this method

Reply:
One of the usages of this method is to toggle the segment state, which can also be a segment operation though.